frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -164,34 +166,43 @@ class Gallery extends React.Component {
     // app crash telling it cannot render a Link outside a router
     // see https://github.com/tracim/tracim/issues/1637
     // GLOBAL_dispatchEvent({type: 'setBreadcrumbs', data: {breadcrumbs: breadcrumbsList}})
-    return breadcrumbsList
+    this.setState({ breadcrumbsList, breadcrumbsLoaded: !includeFile })
   }
 
-  loadFolderDetail = async (workspaceId, folderId) => {
+  loadFolderDetailAndParentsDetails = async (workspaceId, folderId) => {
     const { state, props } = this
 
+    const folderDetail = {}
+
     let fetchContentDetail = await handleFetchResult(
       await getFolderDetail(state.config.apiUrl, workspaceId, folderId)
     )
 
     switch (fetchContentDetail.apiResponse.status) {
       case 200:
-        const fileName = fetchContentDetail.body.filename
-        let folderParentsId = fetchContentDetail.body.parent_id ? [fetchContentDetail.body.parent_id] : []
+        folderDetail.fileName = fetchContentDetail.body.filename
+        folderDetail.folderParentsId = fetchContentDetail.body.parent_id ? [fetchContentDetail.body.parent_id] : []
+
+        let fetchLoopCondition = fetchContentDetail.body.parent_id !== null
+        while (fetchLoopCondition) {
+          const prevParentId = fetchContentDetail.body.parent_id
 
-        while (fetchContentDetail.body.parent_id !== null) {
           fetchContentDetail = await handleFetchResult(
-            await getFolderDetail(state.config.apiUrl, workspaceId, fetchContentDetail.body.parent_id)
+            await getFolderDetail(state.config.apiUrl, workspaceId, prevParentId)
           )
           if (fetchContentDetail.apiResponse.status === 200) {
-            folderParentsId.push(fetchContentDetail.body.parent_id)
+            if (fetchContentDetail.body.parent_id === null || prevParentId === fetchContentDetail.body.parent_id) {
+              fetchLoopCondition = false
+            } else {
+              folderDetail.folderParentsId.push(fetchContentDetail.body.parent_id)
+            }
           } else {
             this.sendGlobalFlashMessage(props.t('Error while loading folder detail'))
             break
           }
         }
 
-        return { fileName, folderParentsId }
+        return folderDetail
       default: this.sendGlobalFlashMessage(props.t('Error while loading folder detail'))
```

Comment:
default case doesn't return anything, in this case, function will return undefined. Even you handle the case where you call that function, it is usually a bad practice